Use the owned EN7573 controller and Q1000K optical profile.

The pinctrl provider and checked PBUS setup replace reference-board SCU/GPIO
writes. Keep the existing path for other boards.

--- a/xpon-en757x/xpon_phy_10g/src/en7581.c
+++ b/xpon-en757x/xpon_phy_10g/src/en7581.c
@@ -463,6 +463,9 @@
 
 int en7581_set_trans_setting(void)
 {
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+	return q1000k_phy_board_profile();
+#else
 	UINT32 read_data = 0;
 	UINT32 write_data = 0;
 	int ret;
@@ -653,6 +656,7 @@
 	}
 	
 	return ret;
+#endif
 }
 
 
@@ -3121,6 +3125,7 @@
 	PON_PHY_PRINT(PHY_MSG_TRACE,"[%s:%d]\r\n",__FUNCTION__,__LINE__);
 
 	
+#if !(defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583))
 #ifdef TCSUPPORT_CPU_ARMV8 
 		/* switch GPIO to XPON mode*/		
 		read_data = GET_IOMUX_CTRL_3(); //GPIO_PON_MODE	
@@ -3154,6 +3159,7 @@
 	IO_SREG(TOP_CSR_GPIO_SHARE, read_data);
 #endif
 
+#endif
 	//enable PCS RX control 	david 20170124
 	read_data=IO_GPHYREG(EN7581_XGPON_PHY_XG_PON_RX_SYNC_CTRL);
 	PON_PHY_PRINT(PHY_MSG_ERR,"Read XGPON_RX_SYNC_CTRL=0x%x \n",read_data);
--- a/xpon-en757x/xpon_phy_10g/src/i2c.c
+++ b/xpon-en757x/xpon_phy_10g/src/i2c.c
@@ -1,3 +1,6 @@
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+#include <q1000k_phy_api.h>
+#endif
 
 #include <linux/version.h>
 #include <linux/kernel.h>
@@ -1597,6 +1600,9 @@
 
 int is_en7572_7573(void) 
 {
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+	return q1000k_phy_controller_check() == 0;
+#else
 
 	UINT8 ptr[2];
 	int ret;
@@ -1626,6 +1632,7 @@
 
 	return ret;
 	
+#endif
 }
 
 
@@ -1978,6 +1985,9 @@
 
 int  phy_trans_model_setting(void)
 {
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+	return q1000k_phy_board_profile();
+#else
 
 	UINT8 ptr[4];
 	
@@ -2127,6 +2137,7 @@
 			return PHY_FAILURE;
 		
 	}
+#endif
 }
 
 int phy_trans_task_wait(void)
--- a/xpon-en757x/xpon_phy_10g/src/phy_tx.c
+++ b/xpon-en757x/xpon_phy_10g/src/phy_tx.c
@@ -1,3 +1,6 @@
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+#include <q1000k_phy_api.h>
+#endif
 /*
 * File Name: phy_tx.c
 * Description: .
@@ -125,6 +128,9 @@
 
 void phy_tx_ctl(unchar val)
 {
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+	q1000k_phy_trans_power(val);
+#else
 
 #if ASIC_SERDES
 	if(PHY_ENABLE == val){ /* enable tx power should turn off tx disable */
@@ -222,6 +228,7 @@
 
 #endif
 
+#endif
 }
 
 
@@ -239,6 +246,9 @@
 ******************************************************************************/
 int phy_trans_power_switch(unchar trans_switch)
 {
+#if defined(TCSUPPORT_CPU_EN7581) && !defined(TCSUPPORT_CPU_AN7583)
+	return q1000k_phy_trans_power(trans_switch);
+#else
 
 	uint read_data = 0;
 	uint write_data = 0 ;
@@ -326,6 +336,7 @@
 		return PHY_FAILURE;
 	}
 
+#endif
 }
 
 void phy_tx_power_config(unchar enable){
